backend/sirius-web-spring-collaborative/src/main/java/org/eclipse/sirius/web/spring/collaborative/projects/EditingContextEventProcessorRegistry.java

```diff
@@ -86,35 +94,54 @@ public EditingContextEventProcessorRegistry(IEditingContextSearchService editing
+                    .map(EditingContextEventProcessorEntry::getEditingContextEventProcessor);
+            if (optionalEditingContextEventProcessor.isEmpty()) {
                 Optional<IEditingContext> optionalEditingContext = this.editingContextSearchService.findById(editingContextId);
                 if (optionalEditingContext.isPresent()) {
                     IEditingContext editingContext = optionalEditingContext.get();
-                    editingContextEventProcessor = new EditingContextEventProcessor(editingContext, this.editingContextPersistenceService, this.applicationEventPublisher, this.objectService,
+
+                    var editingContextEventProcessor = new EditingContextEventProcessor(editingContext, this.editingContextPersistenceService, this.applicationEventPublisher, this.objectService,
                             this.editingContextEventHandlers, this.representationEventProcessorComposedFactory);
-                    this.editingContextEventProcessors.put(editingContextId, editingContextEventProcessor);
+                    Disposable subscription = editingContextEventProcessor.canBeDisposed().delayElements(Duration.ofSeconds(30)).subscribe(canBeDisposed -> {
+                        // We will wait for 30s before trying to dispose the editing context event processor
+                        // We will check if the editing context event processor is still empty
+                        if (canBeDisposed.booleanValue() && editingContextEventProcessor.getRepresentationEventProcessors().isEmpty()) {
+                            this.disposeEditingContextEventProcessor(editingContextId);
+                        } else {
+                            this.logger.trace("Stopping the disposal of the editing context"); //$NON-NLS-1$
+                        }
+                    });
+
+                    var editingContextEventProcessorEntry = new EditingContextEventProcessorEntry(editingContextEventProcessor, subscription);
+                    this.editingContextEventProcessors.put(editingContextId, editingContextEventProcessorEntry);
+
+                    optionalEditingContextEventProcessor = Optional.of(editingContextEventProcessor);
                 }
             }
-            optionalEditingContextEventProcessor = Optional.of(editingContextEventProcessor);
         }
 
         return optionalEditingContextEventProcessor;
     }
 
     @Override
-    public void dispose(UUID editingContextId) {
-        // @formatter:off
-        Optional.ofNullable(this.editingContextEventProcessors.remove(editingContextId))
-                .ifPresent(EditingContextEventProcessor::dispose);
-        // @formatter:on
+    public void disposeEditingContextEventProcessor(UUID editingContextId) {
+        Optional.ofNullable(this.editingContextEventProcessors.remove(editingContextId)).ifPresent(entry -> {
+            entry.getDisposable().dispose();
+            entry.getEditingContextEventProcessor().dispose();
```

Comment:
Same remark as for the other, this should be the responsibility of the entry itself to know what/how to dispose and in which order.